### DISCUSSION TITLE
How to resolve the full json schema for all used plugins?

### DISCUSSION THREAD

**Discussion by vymarkov** [eslint/eslint#15132] [Q&A]
Hi!

I would like to find a method to generate the full JSON schema for used none builtin rules, due to a bunch of different rules, it will be convenient to have the IDE autocomplete while editing the ESLint config file inside IDE. I found that ESLint doesn't provide any public API to resolve JSON schema for none builtin rules and it's hard to find private methods to do it. I would be grateful for help on this. 

It would be nice to provide API for this purpose therefore it makes it possible to supercharge IDE, such VSCode to make fewer errors while working on ESLint config. 

Thanks!

**Comment by nzakas** [comment]
I’m not sure I understand what you’re asking for. Can you explain more about what you are trying to do? And how are you doing this for built in rules right now?

FWIW, you can always just `require` a plugin package and pull data from the rules directly.
